pykafka/balancedconsumer.py

```diff
@@ -398,6 +400,10 @@ def _set_watches(self):
             self._zookeeper, self._consumer_id_path,
             self._consumers_changed
         )
+
+        self._zookeeper.add_listener(self._zookeeper_state_changed)
+        self._zookeeper_connected = self._zookeeper.state == KazooState.CONNECTED
```

Comment:
If the zookeeper is from outside, it may have been started. `self._zookeeper_state_changed` would not be called. And if we not set `self._zookeeper_connected` here, `consume()` will raise an exception.